include/s3select_functions.h

```diff
@@ -116,6 +117,8 @@ enum class s3select_func_En_t {ADD,
                                IS_NOT_NULL,
                                IN,
                                LIKE,
+                               LIKE_ESCAPE,
```

Comment:
one function could be enough (several options in constructor)

in _fn_like constructor should check the following options 
1.incase escape-expr is NULL , use default escape
2.incase like-expr is (variable*) then its constant string, run transform once on to_string(like-expr)

incase escape is dynamic it need to run transform each time (even if like-expr is constant)

transform(and regex compile) could run once only if both like-expr and escape-expr are constant-string